functorch/_src/python_key.py

```diff
@@ -27,33 +29,73 @@ def decomposition_decorator(f):
         return f
     return decomposition_decorator
 
+class Reduction(Enum):
+    NONE = 0
+    MEAN = 1
+    SUM = 2
+
 @register_decomposition(aten.tanh_backward)
-def tanh_backward_decomposition(out_grad, y):
+def tanh_backward_decomposition(out_grad: Tensor, y: Tensor):
     return aten.sub(out_grad, out_grad * y * y)
 
 @register_decomposition(aten.sigmoid_backward)
-def sigmoid_backward_decomposition(out_grad, y):
+def sigmoid_backward_decomposition(out_grad: Tensor, y: Tensor):
     return out_grad * (y * (1 - y))
 
-@register_decomposition(aten._s_where)
-def _s_where_decomposition(a, b, c):
-    return aten.where(a, b, c)
-
 @register_decomposition(aten.detach)
-def detach_decomposition(x):
+# This is valid if we always sit behind autograd
+def detach_decomposition(x: Tensor):
```

Comment:
Actually I'm confused: why do you see detach() in `__torch_dispatch__` if we sit underneath autograd?

Reply:
I think the autograd engine calls detach. And it's not a composite op.